src/extract_gpuinfo_intel.c

```diff
@@ -350,42 +354,41 @@ void gpuinfo_intel_refresh_dynamic_info(struct gpu_info *_gpu_info) {
 
   RESET_ALL(dynamic_info->valid);
 
-  nvtop_device *card_dev_copy;
-  const char *syspath;
-  nvtop_device_get_syspath(gpu_info->card_device, &syspath);
-  nvtop_device_new_from_syspath(&card_dev_copy, syspath);
+  nvtop_device *intel_dev_auto = gpu_info->driver == DRIVER_XE ? gpu_info->driver_device : gpu_info->card_device;
```

Comment:
I'll add a patch to re-introduce the copy. The problem is that libsystemd and libudev may/will cache these values so we need to get create a new device and query the sysattr to get up-to-date values ([see libsystemd implementation](https://github.com/systemd/systemd/blob/3f24fa57df552accc2a6f9ab4d36724ba7227eff/src/libsystemd/sd-device/sd-device.c#L2309))

Reply:
I guess calling `sd_device_set_sysattr_value` with NULL should also do the trick to clear the cache